modules/processors/multidomain_dataset_processor.py

```diff
@@ -1,15 +1,101 @@
-from ..dataset_processor import *
+from ..dataset_processor import Processor
+from modules.processors.utils import chunk_text, listify_label
 import datasets
 import json
+import zipfile
+import random
 
 from tqdm import tqdm
-from hydra.utils import instantiate
 import requests  
 import pandas as pd
+import os
 
 from urllib.parse import unquote
 
 
+class BIOASQ12B(Processor):
+    """ 
+    BIOASQ Benchmark from bioasq challenge source, year 2024 task B (12B)
+    To get a larger training set we merge the official train and validation sets and fix the validation size to 1200 and train size to the rest (= 4189 rows)
+    We then discard all 'summary' question types from the validation set yielding a final val set with 940 rows
+    """
+
+    def __init__(self, train_path, dev_path, *args, **kwargs):
+        self.dataset_name = 'BIOASQ12B'
+        self.train_path = train_path
+        self.dev_path = dev_path
+        super().__init__(*args, **kwargs, dataset_name=self.dataset_name)
+
+    def process(self):
+        seed = 42
+        if self.split not in ["train", "dev"]:
+            raise ValueError("split should be 'train' or 'dev'")
+        all_data = []
+        with zipfile.ZipFile(self.train_path, 'r') as z:
+            with z.open('BioASQ-training12b/training12b_new.json') as json_file:
```

Comment:
Not sure anyone else can get this dataset? ie How can I get this file? Should we upload it to HF? 
BioASQ-training12b/training12b_new.json'

Reply:
Done